postgres/client.js

```diff
@@ -1,39 +1,60 @@
 'use strict';
 
-const bluebird = require('bluebird'),
-  { Client } = require('pg'),
-  QueryStream = require('pg-query-stream'),
-  { POSTGRES_USER, POSTGRES_PASSWORD, POSTGRES_HOST, POSTGRES_PORT, POSTGRES_DB } = require('../services/constants'),
-  { PUT, GET, DEL, BATCH, READ_STREAM } = require('./queries'),
+const { POSTGRES_USER, POSTGRES_PASSWORD, POSTGRES_HOST, POSTGRES_PORT, POSTGRES_DB } = require('../services/constants'),
   { notFoundError } = require('../services/errors'),
-  { parseOrNot, wrapInObject } = require('../services/utils'),
+  { parseOrNot } = require('../services/utils'),
+  { findSchemaAndTable, wrapJSONStringInObject } = require('../services/utils'),
+  knexLib = require('knex'),
   TransformStream = require('../services/list-transform-stream');
-var client = new Client({
-  user: POSTGRES_USER,
-  host: POSTGRES_HOST,
-  database: POSTGRES_DB,
-  password: POSTGRES_PASSWORD,
-  port: POSTGRES_PORT,
-});
+var knex;
 
 /**
- * Connect to the DB
+ * Connect to the default DB and create the Clay
+ * DB. Once the DB has been made the connection
+ * can be killed and we can try to re-connect
+ * to the Clay DB
  *
- * @return {Promise}
+ * @returns {Promise}
  */
-function connect() {
-  return client.connect();
+function createDBIfNotExists() {
+  const tmpClient = knexLib({
+    client: 'pg',
+    connection: {
+      host: POSTGRES_HOST,
+      user: POSTGRES_USER,
+      password: POSTGRES_PASSWORD,
+      database: 'postgres',
+      port: POSTGRES_PORT
+    }
+  });
+
+  return tmpClient.raw(`CREATE DATABASE ${POSTGRES_DB}`)
```

Comment:
After creating the Database do you also need to run a GRANT command so that `POSTGRES_USER` actually has access to it?

Reply:
Haven't run into the issue yet, but I'll look more closely as we move to staging environments